packages/react/src/__tests__/ReactProfiler-test.internal.js

```diff
@@ -246,30 +151,40 @@ describe(`onRender`, () => {
     // Clear yields in case the current time is read during initialization.
     Scheduler.unstable_clearLog();
 
-    ReactTestRenderer.create(
-      <div>
-        <AdvanceTime />
-        <AdvanceTime />
-        <AdvanceTime />
-        <AdvanceTime />
-        <AdvanceTime />
-      </div>,
-    );
+    await act(() => {
+      ReactNoop.render(
+        <div>
+          <AdvanceTime />
+          <AdvanceTime />
+          <AdvanceTime />
+          <AdvanceTime />
+          <AdvanceTime />
+        </div>,
+      );
+    });
 
     // Restore original mock
     jest.mock('scheduler', () => jest.requireActual('scheduler/unstable_mock'));
 
     // TODO: unstable_now is called by more places than just the profiler.
     // Rewrite this test so it's less fragile.
     if (gate(flags => flags.enableDeferRootSchedulingToMicrotask)) {
-      assertLog(['read current time', 'read current time']);
+      assertLog([
+        'read current time',
+        'read current time',
+        'read current time',
+        'read current time',
+      ]);
     } else {
       assertLog([
         'read current time',
         'read current time',
         'read current time',
         'read current time',
         'read current time',
+        'read current time',
+        'read current time',
+        'read current time',
```

Comment:
These assertions have changed after moving from RTR to ReactNoop

Reply:
This is why we have the TODO above: https://github.com/facebook/react/pull/28407/files#diff-76a6c246baa2e1b7bb2de34aede8bf83b69884efcef1fecab903a7e0053b479fR167-R168